### PR TITLE
Fixed count for unpublished nodes

### PR DESCRIPTION
**Pull request by pcambra**

### CODE REVIEW

File: tests/RestfulListTestCase.test [hunk 1]
```diff
@@ -607,6 +607,42 @@ class RestfulListTestCase extends RestfulCurlBaseTestCase {
...
+    $handler = restful_get_restful_handler('articles');
+    // Set a range for pagination to check only .
+    $handler->setRange(9);
+
+    // Check pagination of first page.
+    $result = $handler->get('', array('page' => 1));
+    $output = drupal_json_decode($handler->format($result));
+    $this->assertEqual(count($result), 9, 'Number of results displayed');
+    $this->assertEqual(count($result), $handler->getTotalCount(), 'Count of results takes into account unpublished nodes');
+    $this->assert(empty($output['next']), '"Next" link does not exist on the first page.');
```
Comment (e0ipso) [inline tests/RestfulListTestCase.test]: Maybe `$this->assert` :arrow_right: `$this->assertTrue`?

**Comment by e0ipso** [comment]
Awesome! Special thanks for the test. Just a minor comment before merging.

**Comment by e0ipso** [comment]
I'll merge this with the following commit message, if you think it's appropriate:

```
[BUGFIX] Match default NODE_PUBLISHED filter for count query

Add the filter for NODE_PUBLISHED to the count query by default, just like we are doing it for the regular query. Failing to do this leads to pagination errors.
```

**Comment by pcambra** [comment]
Looks fine, fixed the assertion

**Comment by e0ipso** [comment]
Merged in dc53c14522da8a512100c8c047531d16c8c38b8d.

@pcambra thanks for contributing!

**Comment by e0ipso** [comment]
This was ported to 7.x-2.x in https://github.com/RESTful-Drupal/restful/pull/730. Removing tag.

**Pull request closed without merging**